drivers/ma_reports/app/models/ma_reports/csg_engage/concerns/has_report_status.rb

```diff
@@ -0,0 +1,45 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module MaReports::CsgEngage::Concerns::HasReportStatus
+  extend ActiveSupport::Concern
+  included do
+    [
+      :completed,
+      :failed,
+      :in_progress,
+      :created,
+      :unknown,
+    ].each do |status_value|
+      define_method(:"#{status_value}?") { status == status_value }
+    end
+
+    def status
+      return :completed if completed_at.present?
+      return :failed if failed_at.present?
+      return :in_progress if started_at.present?
+      return :created if created_at.present?
+
+      return :unknown
+    end
+
+    def not_started?
+      return false if completed? || failed? || in_progress? || unknown?
+
+      true
+    end
```

Comment:
This feels really hard to reason about, could it just be 
```
def started?
  started_at.present?
end
```

and then:
```
def not_started?
  ! started?
end
```
if you need the inverse without calling `!` on it?